### ISSUE TITLE
Adding  'upload_mimes' filter should generate a warning 

### ISSUE THREAD

**Issue by gudmdharalds** [Automattic/VIP-Coding-Standards#87]
For instance the following should generate a warning:

```
function abc_upload_mime_types( $mime_types = array() ) {
   $mime_types['test'] = 'text/testing123';
   return $mime_types;
}

add_filter( 'upload_mimes', 'abc_upload_mime_types' );
```
